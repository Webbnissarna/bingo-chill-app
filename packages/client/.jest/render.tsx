import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
