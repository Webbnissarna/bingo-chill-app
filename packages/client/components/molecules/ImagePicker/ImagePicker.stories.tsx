import type { StoryDefault } from "@ladle/react";
import ImagePicker from ".";
import { useState } from "react";

export function Default() {
  const [base64, setBase64] = useState<string | null>(null);

  return (
    <div className="font-text text-snowStorm-0 bg-polarNight-1 p-5">
      <ImagePicker value={base64} onChange={setBase64} />
      <span className="text-[8px] font-monospace block w-96 break-all">
        {base64 ?? "<null>"}
      </span>
    </div>
  );
}

export default {
  title: "Molecules / Image Picker",
} satisfies StoryDefault;
