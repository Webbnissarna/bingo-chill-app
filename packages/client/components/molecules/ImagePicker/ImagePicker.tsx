import { Base64Icon } from "@/components/atoms";
import { LoadingOutlined, PlusOutlined } from "@ant-design/icons";
import { Upload } from "antd";
import type { RcFile, UploadChangeParam, UploadFile } from "antd/es/upload";
import { useState } from "react";

export interface ImagePickerProps {
  value: string | null;
  onChange: (base64: string | null) => void;
}

export default function ImagePicker({
  value,
  onChange,
}: ImagePickerProps): JSX.Element {
  const [isLoading, setIsLoading] = useState(false);

  const processFile = async (info: UploadChangeParam<UploadFile>) => {
    if (!info.file.originFileObj) {
      onChange(null);
    }

    try {
      setIsLoading(true);
      await new Promise((r) => setTimeout(r, 2000));
      const reader = new FileReader();
      reader.readAsDataURL(info.file.originFileObj as RcFile);
      const base64 = await new Promise<string>((resolve) =>
        reader.addEventListener("load", () => resolve(reader.result as string)),
      );
      onChange(base64);
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <div>
      <Upload
        name="avatar"
        listType="picture-circle"
        className="avatar-uploader"
        showUploadList={false}
        beforeUpload={() => undefined}
        customRequest={() => undefined}
        accept=".jpg,.jpeg,.png"
        onChange={processFile}
        disabled={isLoading}
      >
        <div className="font-text text-xs flex flex-col items-center justify-center text-snowStorm-0">
          {value ? (
            <div className="max-h-full max-w-full p-3 rounded-full overflow-hidden aspect-square flex items-center justify-center">
              <Base64Icon src={value} />
            </div>
          ) : isLoading ? (
            <LoadingOutlined />
          ) : (
            <>
              <PlusOutlined />
              <span>Upload</span>
            </>
          )}
        </div>
      </Upload>
    </div>
  );
}
