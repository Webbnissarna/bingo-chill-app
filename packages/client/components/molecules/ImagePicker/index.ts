export { default } from "./ImagePicker";
export * from "./ImagePicker";
