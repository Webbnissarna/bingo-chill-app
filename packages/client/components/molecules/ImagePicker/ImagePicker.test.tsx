import { render } from "../../../.jest/render";
import ImagePicker from ".";

describe("ImagePicker", () => {
  it("renders", () => {
    const { asFragment } = render(<ImagePicker />);

    expect(asFragment()).toMatchSnapshot();
  });
});
