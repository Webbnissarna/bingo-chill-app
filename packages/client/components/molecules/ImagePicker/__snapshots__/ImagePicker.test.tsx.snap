// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker renders 1`] = `
<DocumentFragment>
  <div>
    <span
      class="ant-upload-wrapper avatar-uploader css-dev-only-do-not-override-18iikkb ant-upload-picture-circle-wrapper"
    >
      <div
        class="ant-upload ant-upload-select"
      >
        <span
          class="ant-upload"
          role="button"
          tabindex="0"
        >
          <input
            accept=".jpg,.jpeg,.png"
            style="display: none;"
            type="file"
          />
          <div
            class="font-text text-xs flex flex-col items-center justify-center text-snowStorm-0"
          >
            <span
              aria-label="plus"
              class="anticon anticon-plus"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="plus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                />
                <path
                  d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
                />
              </svg>
            </span>
            <span>
              Upload
            </span>
          </div>
        </span>
      </div>
    </span>
  </div>
</DocumentFragment>
`;
